mongoengine/document.py

```diff
@@ -233,6 +235,28 @@ def _get_collection(cls):
 
         return cls._collection
 
+    @classmethod
+    def _get_timeseries_collection(cls):
+        """Create a new or get an existing timeseries PyMongo collection."""
+        db = cls._get_db()
+        collection_name = cls._get_collection_name()
+        timeseries_opts = cls._meta.get("timeseries")
+
+        if collection_name in list_collection_names(
+            db, include_system_collections=True
+        ):
+            collection = db[collection_name]
+            options = collection.options()
+
+            return collection
+
+        opts = {"expireAfterSeconds": timeseries_opts.pop("expireAfterSeconds", None)}
+        return db.create_collection(
```

Comment:
here why are you creating new collection but the module name is `_get_..`?

Reply:
I chose to name the method _get_timeseries_collection to maintain consistency with the naming conventions used in the original source code. Same as _get_collection method it is responsible for either initializing a new collection or retrieving an existing one, based on the configuration and metadata provided. _get_timeseries_collection is designed to perform a similar role for timeseries collections